rts/idris_utf8.c

```diff
@@ -78,6 +78,36 @@ unsigned idris_utf8_index(char* s, int idx) {
    return top;
 }
 
+char* idris_utf8_advance(char* str, int i) {
```

Comment:
@david-christiansen Sorry for raising my voice here, without being asked. I flagged to issues that IMO lead to buffer over reads below. That can result in serious security issues.

These are made under the assumption that the UTF-8 in the buffer is un-validated. I did not check the call sides.

One way to resolve this is to pass in the buffers size, and make sure you do not pass by its maximum size.